src/contextProviders/accounts-metrics-flow.tsx

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, createContext, useEffect, useState } from "react";
+import { useSearchParams } from "next/navigation";
+import { MetricFlowData } from "../app/functions/universal/getFreeScanSearchParams";
+
+interface SessionProviderProps {
+  children: ReactNode;
+  enabled: boolean;
+  metricsFlowParams: {
+    entrypoint: string;
+    entrypoint_experiment: string;
+    entrypoint_variation: string;
+    form_type: string;
+    service: string;
+  };
+}
+
+type ContextValues = {
+  data: MetricFlowData | null;
+  loading: boolean;
+};
+
+export const AccountsMetricsFlowContext = createContext<ContextValues>({
+  data: null,
+  loading: false,
+});
+
+export const AccountsMetricsFlowProvider = ({
+  children,
+  enabled,
+  metricsFlowParams,
+}: SessionProviderProps) => {
+  const [data, setData] = useState<ContextValues["data"]>(null);
+  const [loading, setLoading] = useState<boolean>(enabled);
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    async function fetchMetricsFlowData() {
+      const updatedSearchParams = new URLSearchParams(searchParams.toString());
```

Comment:
It seems like this will be passing along any/all URL parameters from the Monitor page request to the Monitor API request and subsequently to the FxA `metrics-flow` request, which runs the risk of FxA balking at some random parameters.

IMO the only URL parameters it would make sense to pass along would be UTM parameters, and ideally the UTM parameters used here would be the same as those being passed to the FxA sign-in (e.g. using similar logic as `getFreeScanSearchParams` to look for any appropriate attribution data and fall back to default UTM parameter values if necessary).

Reply:
Makes sense! I’ve restricted the parameters to the ones that are being explicitly passed here in https://github.com/mozilla/blurts-server/pull/4822/commits/e1460963b05b4da7864f245e2687004140942971.

Reply:
It looks like that will always pass `utm_source=product&utm_medium=monitor&utm_campaign=get_free_scan` to `metrics-flow`, which is reasonable, but that isn't consistent with what's being passed to the FxA sign-in (where it first looks for `cookies.attributionsFirstTouch` and then falls back to those default UTM values).

IMO it'd be best to be consistent in both places (either both always pass the preset `get_free_scan` UTM values, or both look for `cookies.attributionsFirstTouch`).

Reply:
Continuing to pass `cookies.attributionsFirstTouch` to the FxA sign-in and not mixing UTM parameters makes sense to me. I’ve adjusted the PR to only pass `utm_source=product&utm_medium=monitor&utm_campaign=get_free_scan` as a fallback if `cookies.attributionsFirstTouch` is not available in [830072b](https://github.com/mozilla/blurts-server/pull/4822/commits/830072bfd147224745e8eec06aa901291c675351).